### PR TITLE
Allow partial closing from dutch auction

### LINKED ISSUE #365: Allow DutchAuction to partially-close positions

### PR DESCRIPTION
**Pull request by BrendanChou**
fixes #365

### CODE REVIEW

File: contracts/margin/external/ERC20/ERC20Position.sol [hunk 1]
```diff
@@ -330,23 +322,29 @@ contract ERC20Position is
...
      * carry out such an attack.
      *
-     * @param  who  Address of the account to withdraw for
-     * @return      The amount of heldToken withdrawn
+     * @param  who  Array of addresses to withdraw for
+     * @return      The amount of heldToken withdrawn for each address
      */
-    function withdraw(
-        address who
+    function withdrawMultiple(
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20Position.sol]: As per our discussion, you either need to:

- Leave this as it was before
- Change all of these types of functions to only have one version that takes an array, and name it `withdraw`

Either way, this should be done in a separate PR

File: contracts/margin/external/ERC20/ERC20Long.sol [hunk 2]
```diff
@@ -99,31 +92,37 @@ contract ERC20Long is ERC20Position {
...
             balance,
-            totalSupply_,
+            positionBalance,
             positionPrincipal
         );
 
-        uint256 tokenAmount = MathHelpers.getPartialAmount(
-            maxAllowedCloseAmount,
+        // the new close amount should not be higher than what was requested
+        assert(allowedCloseAmount < requestedCloseAmount);
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20Long.sol]: should this be `<=`?

Comment (BrendanChou) [reply contracts/margin/external/ERC20/ERC20Long.sol]: If it were equal, then it would have returned up above
